### ISSUE TITLE
Guava library conflict or 

### ISSUE THREAD

**Issue by Swah** [RenderHeads/UnityPlugin-AVProVideo#1135]
**Describe the issue**
We're not able to build on Android with our current project, see below for logs. We suspect the problem is the same as this one listed here: 
https://github.com/RenderHeads/UnityPlugin-AVProVideo/issues/630

We do have firebase in our unity project, so we suspect the cause is two guava libraries conflicting with each other. As suggested in this post, we tried deleting the included guava-27.1-android.jar file. The project now builds, but crashes on an android device when opening a scene with AvPro components in it
AndroidJavaException: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/common/collect/ImmutableList;
Which we interpret as meaning that the guava library is missing. 

We've also tried adding guava 27.1-android as a dependency through Google's External Dependency Manager for Unity, and removing the .jar from the project, but that gets us the same initial build error. 

What solution is recommended? Ideally the solution would use the External Dependency Manager for Unity in a way that is compatible with other plugins. 

**Your Setup (please complete the following information):**
 - Unity version: 2019.4.35f1
 - AVPro Video version (number and edition (trial/core/ultra/enterprise)): trial [AVPro Video v2.4.1]
 - Operating system version: Windows 11
 - Device model: Android Pixel 4a
 - Video specs (resolution, frame-rate, codec, file size): MP4, but not the issue here

**Logs**
BUILD LOG ERRORS
[Error] [None] Starting a Gradle Daemon, 1 incompatible and 3 stopped Daemons could not be reused, use --status for details
> Configure project :launcher
WARNING: The option setting 'android.bundle.enableUncompressedNativeLibs=false' is experimental and unsupported.
The current default is 'true'.
Exception while marshalling C:\Program Files\Unity\Hub\Editor\2019.4.35f1\Editor\Data\PlaybackEngines\AndroidPlayer\SDK\build-tools\30.0.2\package.xml. Probably the SDK is read-only
Exception while marshalling C:\Prog
...
[Error] [None] Note: C:\Users\franc\UlulabRepos\Numbers5\Temp\gradleOut\unityLibrary\src\main\java\com\unity3d\player\UnityPlayerActivity.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
D8: Cannot fit requested classes in a single dex file (# methods: 67473 > 65536)
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':launcher:transformDexArchiveWithExternalLibsDexMergerForRelease'.
> com.android.builder.dexing.DexArchiveMergerExc
...
[Exception] [None] CommandInvokationFailure: Gradle build failed. 
[Error] [None] Build error: Build failed with 4 error(s)

LOGCAT CRASH MESSAGE AFTER REMOVING GUAVA JAR
03-22 09:03:24.841  1762  2066 I ActivityManager: Start proc 11062:com.google.android.euicc/u0a160 for service {com.google.android.euicc/com.android.euicc.service.EuiccServiceImpl}
03-22 09:04:46.431  1762  2066 I ActivityManager: Start proc 11181:com.discord/u0a228 for broadcast {com.discord/com.google.firebase.iid.FirebaseInstanceIdReceiver}
03-22 09:04:56.290 10639 10663 I Unity   : s_Interface UnityEngine.AndroidJavaObject
03-22 09:04:56.290 10639 10663 I Unity   : RenderHeads.Media.AVProVideo.AndroidMediaPlayer:.ctor(OptionsAndroid)
03-22 09:04:56.290 10639 10663 I Unity   : RenderHeads.Media.AVProVideo.MediaPlayer:CreateMediaPlayer(OptionsAndroid)
03-22 09:04:56.290 10639 10663 I Unity   : RenderHeads.Media.AVProVideo.MediaPlayer:CreateMediaPlayer()
03-22 09:04:56.290 10639 10663 I Unity   : RenderHeads.Media.AVProVideo.MediaPlayer:Initialise()
03-22 09:04:56.290 10639 10663 I Unity   : RenderHeads.Media.AVProVideo.MediaPlayer:Start()
03-22 09:04:56.290 10639 10663 I Unity   :
03-22 09:04:56.290 10639 10663 I Unity   : (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 39)
03-22 09:04:56.290 10639 10663 I Unity   :
03-22 09:04:56.315 10639 10663 E Unity   : AndroidJavaException: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/common/collect/ImmutableList;
03-22 09:04:56.315 10639 10663 E Unity   : java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/common/collect/ImmutableList;
03-22 09:04:56.315 10639 10663 E Unity   :      at com.google.android.exoplr2avp.upstream.DefaultBandwidthMeter.<clinit>(DefaultBandwidthMeter.java:45)
03-22 09:04:56.315 10639 10663 E Unity   :      at com.google.android.exoplr2avp.upstream.DefaultBandwidthMeter.getSingletonInstance(Unknown Source:0)
03-22 09:04:56.315 10639 10663 E Unity   :      at com.renderheads.AVPro.Video.Player_ExoPlayer.InitialisePlayer(Player_ExoPlayer.java:48)
03-22 09:04:56.315 10639 10663 E Unity   :      at com.renderheads.AVPro.Video.Player_Base.Initialise(Player_Base.java:57)
03-22 09:04:56.315 10639 10663 E Unity   :      at com.renderheads.AVPro.Video.Manager.CreatePlayer(Manager.java:30)
03-22 09:04:56.315 10639 10663 E Unity   :      at com.unity3d.player.UnityPlayer.nativeRender(Native Method)
03-22 09:04:56.315 10639 10663 E Unity   :      at com.unity3d.player.UnityPlayer.access$300(Unknown Source:0)
03-22 09:04:56.315 10639 10663 E Unity   :      at com.unity3d.player.UnityPlayer$e$1.handleMessage(Unknown Source:95)
03-22 09:04:56.315 10639 10663 E Unity   :      at android.os.Handler.dispatchMessage(Handler.java:102)
03-22 09:04:56.315 10639 10663 E Unity   :      at android.os.Looper.loopOnce(Looper.java:201)
03-22 09:04:56.315 10639 10663 E Unity   :      at android.os.Looper.loop(Looper.java:288)
03-22 09:04:56.315 10639 10663 E Unity   :      at com.unity3
03-22 09:04:56.493  1762  7900 I ActivityManager: Process com.ululab.numbers (pid 10639) has died: prcp TOP

### Please DO NOT LINK / ATTACH YOUR PROJECT FILES HERE
Instead email the link to us unitysupport@renderheads.com

**Comment by kahnivore** [comment]
Hi there,

Could you try the following and let us know if this solves it: 

1. Delete AVProVideo\Runtime\Plugins\Android\guava-27.1-android.jar;

2. Player Settings -> Publishing Settings -> Build -> Custom Main Gradle Templete Checked;

3. Edit the Assets\Plugins\Android\mainTemplate.gradle file and modify dependencies to:

Code (JavaScript):
 dependencies {
   implementation fileTree(dir: 'libs', include: ['*.jar'])
   implementation("com.google.guava:guava:31.0.1-android")
   **DEPS**}

4. Continue to compile, if the compilation reports an error, restart Unity and try again;

**Comment by Swah** [comment]
Thank you for your reply. We just found a similar way to do this, but using Google's External Dependency Manager for Unity. Noting it here for other devs:

1. added a file Asets/AVProVideo/AVProDependencies.xml with 
`<dependencies>

  
  <androidPackages>
    <androidPackage spec="com.google.guava:guava:27.1-android">
    </androidPackage>
  </androidPackages>
</dependencies>`
2. Changed Minimum API Level to Android 5.0

I noticed you've used Guava version 31 in your suggestion, should we also be using that newer version instead of version 27.1 that is provided with the plugin?
